device_doctor/lib/src/ios_device.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' show LineSplitter;
+
+import 'package:meta/meta.dart';
+
+import 'device.dart';
+import 'process_helper.dart';
+
+class IosDeviceDiscovery implements DeviceDiscovery {
+  factory IosDeviceDiscovery() {
+    return _instance ??= IosDeviceDiscovery._();
+  }
+
+  IosDeviceDiscovery._();
+
+  static IosDeviceDiscovery _instance;
+
+  @override
+  Future<List<Device>> discoverDevices({int retriesDelayMs = 10000}) async {
+    // Warm up device list. This is to ensure slow devices are discovered.
+    // https://github.com/flutter/flutter/issues/64753
+    // await eval('xcrun', <String>['xcdevice', 'list', '--timeout', '10']);
+    final List<String> iosDeviceIds = LineSplitter.split(await eval('idevice_id', <String>['-l'])).toList();
+    return iosDeviceIds.map((String id) => IosDevice(deviceId: id)).toList();
+  }
+
+  @override
+  Future<Map<String, HealthCheckResult>> checkDevices() async {
+    final Map<String, HealthCheckResult> results = <String, HealthCheckResult>{};
+    for (Device device in await discoverDevices()) {
+      results['ios-device-${device.deviceId}'] = HealthCheckResult.success();
+    }
+    return results;
+  }
+
+  @override
+  Future<void> recoverDevice() async {
+    for (Device device in await discoverDevices()) {
+      await device.recover();
+    }
+  }
+}
+
+/// iOS device.
+class IosDevice implements Device {
```

Comment:
Are we redefining the Device interface? do we need an interface? e.g are we expecting multiple implementations of this interface?